### PR TITLE
Fawn critter

### PR DESCRIPTION
**Pull request by theQuazz**
This is just a demo, I'm not sure that we'd actually merge this

### CODE REVIEW

File: wurst/systems/units/hostiles/Fawn.wurst [hunk 1]
```diff
@@ -15,9 +23,32 @@ public class Fawn extends BonyAnimal
...
+        let pos = getPos()
+        let acorn = findNearestItem(pos, 300, i -> i.getTypeId() == ITEM_ACORN)
+        if acorn == null
+            return
+
+        let acornPos = acorn.getPos()
+        if acornPos.distanceTo(pos) < 50
+            acorn.remove()
+            new DummyCaster
+            ..owner(players[0])
```
Comment (jonathanmclaus) [inline wurst/systems/units/hostiles/Fawn.wurst]: How does this work? What player gets the unit?

**Comment by Marsunpaisti** [comment]
How does DummycasterFixed differ from ITTDummycaster? Or is this just a rename

**Comment by theQuazz** [comment]
@Marsunpaisti this library is in the standard library (not just my own). However, it was bugged, so I fixed it and commit it, but we'll switch to the main standard library version once its fixed there.

File: wurst/utils/UnitUtils.wurst [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+public function findNearestUnit(vec2 pos, real range, filterfunc filter) returns unit
+    GroupEnumUnitsInRange(ENUM_GROUP, pos.x, pos.y, range, filter)
+    unit nearest = null
+    var bestDist = REAL_MAX
+    for u from ENUM_GROUP
+        let distSq = pos.distanceToSq(u.getPos())
+        if distSq < bestDist
+            nearest = u
+            bestDist = distSq
+    return nearest
```
Comment (Marsunpaisti) [inline wurst/utils/UnitUtils.wurst]: Did you perhaps miss this

https://gyazo.com/0326ee81b1359bc4f7c0182db45a4118

File: wurst/systems/units/hostiles/RearingAnimal.wurst [hunk 3]
```diff
@@ -24,7 +24,7 @@ public abstract class RearingAnimal extends BonyAnimal
         return drops
 
     override function onDeath()
-        if getAnimalGreenLight(getUnit())
+        if true // getAnimalGreenLight(getUnit())
             createChild()
 
         super.onDeath()
```
Comment (Marsunpaisti) [inline wurst/systems/units/hostiles/RearingAnimal.wurst]: Is this supposed to be always true

File: wurst/systems/spells/ElkCritter.wurst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package ElkCritter
+import AbilityObjEditing
+import ObjectIdGenerator
+
+public let SPELL_ELK_CRITTER_ID = compiletime(ABIL_ID_GEN.next())
+
+@compiletime function createSpellElkCritter()
+    new AbilityDefinitionMechanicalCritter(SPELL_ELK_CRITTER_ID)
```
Comment (Marsunpaisti) [inline wurst/systems/spells/ElkCritter.wurst]: No additional config for the spell? Not sure what the spell default object looks like but maybe add 0 cooldown, 0 manacost, 99999 range, no art etc

File: wurst/systems/units/hostiles/Fawn.wurst [hunk 5]
```diff
@@ -17,5 +45,40 @@ public class Fawn extends BabyAnimal
...
+            return
+
+        let pos = fawn.getPos()
+        let acorn = findNearestItem(pos, 300, i -> i.getTypeId() == ITEM_ACORN)
+        if acorn == null
+            return
+
+        let acornPos = acorn.getPos()
+        if acornPos.distanceTo(pos) > 50
+            fawn.issuePointOrderById(Orders.move, acornPos)
```
Comment (Marsunpaisti) [inline wurst/systems/units/hostiles/Fawn.wurst]: use squared distance

File: wurst/systems/spells/ElkCritter.wurst (hunk 4 above)
Comment (theQuazz) [reply wurst/systems/spells/ElkCritter.wurst]: @Marsunpaisti none of those things are configurable for this spell lol

File: wurst/systems/units/hostiles/Fawn.wurst (hunk 5 above)
Comment (theQuazz) [reply wurst/systems/units/hostiles/Fawn.wurst]: @Marsunpaisti then I have to square my real value as well, which I'm not inclined to do. I think `distanceSq` is fine when you're comparing two distances to one another but when you're comparing to a scalar it is less readable.

File: wurst/systems/units/hostiles/RearingAnimal.wurst (hunk 3 above)
Comment (theQuazz) [reply wurst/systems/units/hostiles/RearingAnimal.wurst]: @Marsunpaisti updated, refresh

File: wurst/utils/UnitUtils.wurst (hunk 2 above)
Comment (theQuazz) [reply wurst/utils/UnitUtils.wurst]: @Marsunpaisti I didn't, that method returns the unit in a closure which I didn't want (and is unnecessary)

**Pull request merged**